lib/resque/server.rb

```diff
@@ -83,6 +85,8 @@ def redis_get_value_as_array(key, start=0)
         case Resque.redis.type(key)
         when 'none'
           []
+        when 'hash'
+          Resque.redis.hgetall(key).to_a[start..(start + 20)]
```

Comment:
`HGETALL` has a noticeable performance problem when the size of the hash is large.
We do not expect to use such large hashes in Resque, so I use `HGETALL` here.